File: src/main/java/com/lifecycle/components/entities/Extent.java
package com.lifecycle.components.entities;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Extent {
	private String reference;
	
	@JsonProperty("x min")
	private double x_min;
	
	@JsonProperty("x max")
	private double x_max;
	
	@JsonProperty("y min")
	private double y_min;
	
	@JsonProperty("y max")
	private double y_max;
	
	public Extent() {
		super();
	}

	public String getReference() {
		return reference;
	}

	public void setReference(String reference) {
		this.reference = reference;
	}

	public double getX_min() {
		return x_min;
	}

	public void setX_min(double x_min) {
		this.x_min = x_min;
	}

	public double getX_max() {
		return x_max;
	}

	public void setX_max(double x_max) {
		this.x_max = x_max;
	}

	public double getY_min() {
		return y_min;
	}

	public void setY_min(double y_min) {
		this.y_min = y_min;
	}

	public double getY_max() {
		return y_max;
	}

	public void setY_max(double y_max) {
		this.y_max = y_max;
	}
}

File: src/main/resources/4thYrProjectWebsite/app-framework/data_structures/metadata/extent.js
'use strict';

import JsonObject from '../../base/json-object.js';

/**
 * Metadata extent element
 * @module metadata/extent
 * @extends JsonObject
 */
export default class Extent extends JsonObject { 
	
	get reference() { return this.json["reference"]; }
	
	get x_min() { return this.json["x min"]; }
	
	get y_min() { return this.json["y min"]; }
	
	get x_max() { return this.json["x max"]; }
	
	get y_max() { return this.json["y max"]; }
	/*
	static make(reference, x_min, y_min, x_max, y_max) {
		return new Extent({
			"reference": reference,
			"x min": x_min,
			"y min": y_min,
			"x max": x_max,
			"y max": y_max
		});
	}
	*/
	static make(reference, x_min, y_min, x_max, y_max) {
		return {
			"reference": reference,
			"x min": x_min,
			"y min": y_min,
			"x max": x_max,
			"y max": y_max
		};
	}
}